sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ToNumberParser.scala

```diff
@@ -599,4 +612,227 @@ class ToNumberParser(numberFormat: String, errorOnFail: Boolean) extends Seriali
       Decimal(javaDecimal, precision, scale)
     }
   }
+
+  /**
+   * Converts a decimal value to a string based on the given number format.
+   *
+   * Iterates through the [[formatTokens]] obtained from processing the format string, while also
+   * inspecting the input decimal value.
+   *
+   * @param input the decimal value that needs to be converted
+   * @return the result String value obtained from string formatting
+   */
+  def format(input: Decimal): UTF8String = {
+    val result = new StringBuilder()
+    // These are string representations of the input Decimal value.
+    val (inputBeforeDecimalPoint: String,
+      inputAfterDecimalPoint: String) =
+      formatSplitInputBeforeAndAfterDecimalPoint(input).getOrElse(
```

Comment:
How about just `input.toString.trim("0").split(".")`? It must be a valid decimal string and we don't need complicated logic to parse it.

Reply:
nvm, `formatSplitInputBeforeAndAfterDecimalPoint` kinds of normalize the decimal string according to the format pattern.

Reply:
but `input.toString.trim("0").split(".")` can be used in `formatSplitInputBeforeAndAfterDecimalPoint` to simplify the code.

Reply:
Good idea, I was able to use this to simplify the code in `formatSplitInputBeforeAndAfterDecimalPoint`.